Sequences/MRML/vtkMRMLNodeSequencer.h

```diff
@@ -45,7 +45,7 @@ class VTK_SLICER_SEQUENCES_MODULE_MRML_EXPORT vtkMRMLNodeSequencer
 public:
 
   /// Information and methods that are needed for efficient recording and replay of sequences.
-  class NodeSequencer
+  class VTK_SLICER_SEQUENCES_MODULE_MRML_EXPORT NodeSequencer
```

Comment:
I thought that symbols like `vtkMRMLNodeSequencer::NodeSequencer` would be available by default without the need to explicitly export the inner classes or structs.

Reply:
I tried to subclass NodeSequencer in another module and ended up getting a number of errors:

~~~
vtkSlicerVideoIOLogic.obj : error LNK2019: unresolved external symbol "public: __cdecl vtkMRMLNodeSequencer::NodeSequencer::NodeSequencer(void)" (??0NodeSequencer@vtkMRMLNodeSequencer@@QEAA@XZ) referenced in function "public: __cdecl StreamingVolumeNodeSequencer::StreamingVolumeNodeSequencer(void)" (??0StreamingVolumeNodeSequencer@@QEAA@XZ)
vtkSlicerVideoIOLogic.obj : error LNK2019: unresolved external symbol "public: virtual __cdecl vtkMRMLNodeSequencer::NodeSequencer::~NodeSequencer(void)" (??1NodeSequencer@vtkMRMLNodeSequencer@@UEAA@XZ) referenced in function "int `public: __cdecl StreamingVolumeNodeSequencer::StreamingVolumeNodeSequencer(void)'::`1'::dtor$0" (?dtor$0@?0???0StreamingVolumeNodeSequencer@@QEAA@XZ@4HA)
vtkSlicerVideoIOLogic.obj : error LNK2001: unresolved external symbol "public: virtual class vtkMRMLNode * __cdecl vtkMRMLNodeSequencer::NodeSequencer::DeepCopyNodeToScene(class vtkMRMLNode *,class vtkMRMLScene *)" (?DeepCopyNodeToScene@NodeSequencer@vtkMRMLNodeSequencer@@UEAAPEAVvtkMRMLNode@@PEAV3@PEAVvtkMRMLScene@@@Z)
vtkSlicerVideoIOLogic.obj : error LNK2001: unresolved external symbol "public: virtual class vtkIntArray * __cdecl vtkMRMLNodeSequencer::NodeSequencer::GetRecordingEvents(void)" (?GetRecordingEvents@NodeSequencer@vtkMRMLNodeSequencer@@UEAAPEAVvtkIntArray@@XZ)
vtkSlicerVideoIOLogic.obj : error LNK2001: unresolved external symbol "public: virtual class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __cdecl vtkMRMLNodeSequencer::NodeSequencer::GetSupportedNodeClassName(void)" (?GetSupportedNodeClassName@NodeSequencer@vtkMRMLNodeSequencer@@UEAA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@XZ)
vtkSlicerVideoIOLogic.obj : error LNK2001: unresolved external symbol "public: virtual bool __cdecl vtkMRMLNodeSequencer::NodeSequencer::IsNodeSupported(class vtkMRMLNode *)" (?IsNodeSupported@NodeSequencer@vtkMRMLNodeSequencer@@UEAA_NPEAVvtkMRMLNode@@@Z)
vtkSlicerVideoIOLogic.obj : error LNK2001: unresolved external symbol "public: virtual bool __cdecl vtkMRMLNodeSequencer::NodeSequencer::IsNodeSupported(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?IsNodeSupported@NodeSequencer@vtkMRMLNodeSequencer@@UEAA_NAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z)
vtkSlicerVideoIOLogic.obj : error LNK2001: unresolved external symbol "public: virtual void __cdecl vtkMRMLNodeSequencer::NodeSequencer::AddDefaultDisplayNodes(class vtkMRMLNode *)" (?AddDefaultDisplayNodes@NodeSequencer@vtkMRMLNodeSequencer@@UEAAXPEAVvtkMRMLNode@@@Z)
~~~

Adding the export macro fixes this.

Reply:
On Windows, by default all symbols are private, unless exported by this macro.